src/openfermion/ops/_boson_operator.py

```diff
@@ -11,14 +11,8 @@
 #   limitations under the License.
 
 """BosonOperator stores a sum of products of bosonic ladder operators."""
-import numpy
 
-from future.utils import iteritems
-from openfermion.ops import SymbolicOperator
-
-
-class BosonOperatorError(Exception):
-    pass
+from openfermion.ops._symbolic_operator import SymbolicOperator
```

Comment:
Why the shift away from importing from `openfermion.ops`? (No preference, just wondering about the motivation for the change)

Reply:
Importing directly from `ops` from a file inside `ops` is brittle and can cause circular dependencies. This is the reason we need to put imports out of alphabetical order in `ops/__init__.py`. Doing it my way, we won't need to shuffle around imports in `__init__.py` files.

Reply:
It's okay to import from `ops` in the test files though. In fact, test files should just import from the top level, `openfermion`, since that is how we should be encouraging users to write their code.